src/server/hook_func.c

```diff
@@ -5279,142 +5286,176 @@ sync_mom_hookfiles(void *minfo)
+						PBS_EVENTCLASS_REQUEST, LOG_WARNING,
+						"", log_buffer);
+				} else {
+					if (pbs_errno != PBSE_MOM_REJECT_ROOT_SCRIPTS) {
 						snprintf(log_buffer, sizeof(log_buffer),
-							"errno %d: failed to copy hook file %s to %s:%d",
-							pbs_errno, hookfile, minfo_array[i]->mi_host,
+							"successfully sent hook file %s to %s:%d",
+							hookfile, minfo_array[i]->mi_host,
 							minfo_array[i]->mi_port);
-						log_event(PBSEVENT_DEBUG3,
-							PBS_EVENTCLASS_REQUEST, LOG_WARNING,
+						log_event(PBSEVENT_DEBUG,
+							PBS_EVENTCLASS_REQUEST, LOG_INFO,
 							"", log_buffer);
 					} else {
-						if (pbs_errno != PBSE_MOM_REJECT_ROOT_SCRIPTS) {
-							snprintf(log_buffer, sizeof(log_buffer),
-								"successfully sent hook file %s to %s:%d",
-								hookfile, minfo_array[i]->mi_host,
-								minfo_array[i]->mi_port);
-							log_event(PBSEVENT_DEBUG,
-								PBS_EVENTCLASS_REQUEST, LOG_INFO,
-								"", log_buffer);
-						} else {
-							snprintf(log_buffer, sizeof(log_buffer),
-								"warning: sending hook file %s to %s:%d got rejected (mom's reject_root_scripts=1)",
-								hookfile, minfo_array[i]->mi_host,
-								minfo_array[i]->mi_port);
-							log_event(PBSEVENT_DEBUG3,
-								PBS_EVENTCLASS_REQUEST, LOG_INFO,
-								"", log_buffer);
-						}
-						pact->action &= ~(MOM_HOOK_ACTION_SEND_CONFIG);
-						hook_track_save((mominfo_t *)minfo_array[i],
-							j);
-						action_done++;
+						snprintf(log_buffer, sizeof(log_buffer),
+							"warning: sending hook file %s to %s:%d got rejected (mom's reject_root_scripts=1)",
+							hookfile, minfo_array[i]->mi_host,
+							minfo_array[i]->mi_port);
+						log_event(PBSEVENT_DEBUG3,
+							PBS_EVENTCLASS_REQUEST, LOG_INFO,
+							"", log_buffer);
 					}
-				} /* send config */
+					pact->action &= ~(MOM_HOOK_ACTION_SEND_CONFIG);
+					hook_track_save((mominfo_t *)minfo_array[i],
+						j);
+					action_done++;
+				}
+			} /* send config */
 
-				/* send hook content */
-				if (pact->action & MOM_HOOK_ACTION_SEND_SCRIPT) {
+			/* send hook content */
+			if (pact->action & MOM_HOOK_ACTION_SEND_SCRIPT) {
 
-					snprintf(hookfile, MAXPATHLEN, "%s%s%s",
-						path_hooks, pact->hookname,
-						HOOK_SCRIPT_SUFFIX);
+				snprintf(hookfile, MAXPATHLEN, "%s%s%s",
+					path_hooks, pact->hookname,
+					HOOK_SCRIPT_SUFFIX);
 
-					if ((PBSD_copyhookfile(conn, hookfile, 0, NULL) != 0) &&
-						(pbs_errno != PBSE_MOM_REJECT_ROOT_SCRIPTS)) {
+				if ((phook->event & MOM_EVENTS) == 0) {
+					pact->action &= ~MOM_HOOK_ACTION_SEND_SCRIPT;
+				} if ((PBSD_copyhookfile(conn, hookfile, 0, NULL) != 0) &&
+					(pbs_errno != PBSE_MOM_REJECT_ROOT_SCRIPTS)) {
+					snprintf(log_buffer, sizeof(log_buffer),
+						"errno %d: failed to copy hook file %s to %s:%d",
+						pbs_errno, hookfile, minfo_array[i]->mi_host,
+						minfo_array[i]->mi_port);
+					log_event(PBSEVENT_DEBUG3,
+						PBS_EVENTCLASS_REQUEST, LOG_WARNING,
+						"", log_buffer);
```

Comment:
Instead of using an empty string as the object name, you should use `__func__`

There are several instances of this.

Reply:
I think it is more appropriate to use 'msg_daemonname'  (e.g. Server) as object_name of which the message is related to, rather than the function name it came from. I just checked other instances in the server code where msg_daemonname is used and not __func__. So I'll set it to that.